pageserver/src/tenant/timeline.rs

```diff
@@ -602,13 +603,58 @@ impl From<layer_manager::Shutdown> for GetVectoredError {
     }
 }
 
+pub struct ReadPath {
+    keyspace: KeySpace,
+    lsn: Lsn,
+    path: Vec<(LayerId, KeySpace, Range<Lsn>)>,
```

Comment:
Why use the `LayerId` here? It loses information for the in-memory layers.

Reply:
fixed in https://github.com/neondatabase/neon/pull/10528/commits/809baefa227016258e9dd346d763b91af0aa3118